www/js/services.js

```diff
@@ -102,6 +102,91 @@ angular.module('emission.services', [])
                "Email cancel reported, seems to be an error on android");
         });
     }
+
+    this.dataCollectionSetConfig = function(config) {
+      return new Promise(function(resolve, reject) {
+          resolve(window.cordova.plugins.BEMDataCollection.setConfig(config));
+      });
+    };
+
+    this.dataCollectionGetConfig = function() {
+      return new Promise(function(reslove, reject) {
+          reslove(window.cordova.plugins.BEMDataCollection.getConfig());
+      });
+    };
+    this.serverSyncSetConfig = function(config) {
+      return new Promise(function(resolve, reject) {
+          resolve(window.cordova.plugins.BEMServerSync.setConfig(config));
+      });
+    };
+
+    this.serverSyncGetConfig = function() {
+      return new Promise(function(reslove, reject) {
+          reslove(window.cordova.plugins.BEMServerSync.getConfig());
+      });
+    };
+
+    this.getAccuracyOptions = function() {
+      return new Promise(function(reslove, reject) {
+          reslove(window.cordova.plugins.BEMDataCollection.getAccuracyOptions());
+      });
+    };
+
+    this.getUserEmail = function() {
+      return new Promise(function(reslove, reject) {
+        window.cordova.plugins.BEMJWTAuth.getUserEmail(function(response) {
+            reslove(response);
+        }, function(error) {
+            reject(error);
+        });
+      });
+    };
+
+    this.getState = function() {
+      return new Promise(function(reslove, reject) {
+        window.cordova.plugins.BEMDataCollection.getState(function(response) {
```

Comment:
I am trying to promisify the plugins but for the ones with callback like getState()
https://github.com/e-mission/e-mission-data-collection/blob/master/www/datacollection.js#L48
the success and error callback parameter is passed into exec() and promises depends on the response of the callback so is it possible to make this plugin return a promise?